packages/core/src/components/tag-input/resizeableInput.tsx

```diff
@@ -0,0 +1,50 @@
+/* !
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ */
+
+import React, { forwardRef, useEffect, useRef, useState } from "react";
+
+import { HTMLInputProps } from "../../common";
+
+type ResizeableInputProps = HTMLInputProps;
+export type Ref = HTMLInputElement;
+
+export const ResizeableInput = forwardRef<Ref, ResizeableInputProps>(function ResizeableInput(props, ref) {
+    const [content, setContent] = useState("");
+    const [width, setWidth] = useState(0);
+    const span = useRef<HTMLSpanElement>(null);
+
+    useEffect(() => {
+        if (span.current) {
+            setWidth(span.current.offsetWidth);
+        }
+    }, [content]);
+
+    const { onChange, ...otherProps } = props;
+
+    const changeHandler: React.ChangeEventHandler<HTMLInputElement> = evt => {
+        setContent(evt.target.value);
+        onChange?.(evt);
+    };
+
+    return (
+        <span>
+            <span
+                id="hide"
+                ref={span}
+                style={{
```

Comment:
prefer styling with CSS rather than inline styles. you should create a `_resizableInput.scss` file next to this one and create a new class following our existing naming conventions...

```scss
.#{$ns}-resizable-input-span {
  max-height: 0;
  max-width: 100%;
  min-width: 80px;
  position: absolute;
  white-space: nowrap;
  z-index: -100;
}
```